### PR TITLE
chore(amazon-cognito-identity-js): Adds migration documentation

### PR DESCRIPTION
**Pull request by haverchuck**
#### Description of how you validated changes
Adds migration information for amazon-cognito-identity-js -> Auth v6.

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [X] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazon-cognito-identity-js/README.md [hunk 1]
```diff
@@ -1,3 +1,46 @@
...
+
+The following table provides the Amplify Auth APIs that can be used to satisfy Amazon Cognito Identity SDK use cases.
+
+# Developer Note: Please Update to Amplify v6
+
+We recommend using the Amplify JavaScript library's Auth features in place of the Amazon Cognito Identity SDK. The Amplify JavaScript library offers a modern, fully-typed and performant experience for Auth use cases.
+
+## Amazon Cognito Identity SDK and Amplify JavaScript API Mappings
+
+The following table provides the Amplify Auth APIs that can be used to satisfy Amazon Cognito Identity SDK use cases.
```
Comment (jimblanc) [inline packages/amazon-cognito-identity-js/README.md:15]: Duplicated?

**Review by jimblanc — COMMENTED** [review]
Thanks for doing this! Few nits

File: packages/amazon-cognito-identity-js/README.md [hunk 2]
```diff
@@ -1,3 +1,46 @@
+# Developer Note: Please Update to Amplify v6
+
+We recommend using the Amplify JavaScript library's Auth features in place of the Amazon Cognito Identity SDK. The Amplify JavaScript library offers a modern, fully-typed and performant experience for Auth use cases.
```
Comment (renebrandel) [inline packages/amazon-cognito-identity-js/README.md:3]:
```suggestion
We recommend using the Amplify JavaScript library's Auth features in place of the Amazon Cognito Identity SDK. The Amplify JavaScript library offers a modern, fully-typed, and performant experience for Auth use cases with tree-shaking built-in for bundle size reduction.
```

File: packages/amazon-cognito-identity-js/README.md [hunk 3]
```diff
@@ -1,3 +1,46 @@
...
+
+The following table provides the Amplify Auth APIs that can be used to satisfy Amazon Cognito Identity SDK use cases.
+
+# Developer Note: Please Update to Amplify v6
+
+We recommend using the Amplify JavaScript library's Auth features in place of the Amazon Cognito Identity SDK. The Amplify JavaScript library offers a modern, fully-typed and performant experience for Auth use cases.
+
+## Amazon Cognito Identity SDK and Amplify JavaScript API Mappings
+
+The following table provides the Amplify Auth APIs that can be used to satisfy Amazon Cognito Identity SDK use cases.
```
Comment (renebrandel) [inline packages/amazon-cognito-identity-js/README.md:15]:
```suggestion
```

File: packages/amazon-cognito-identity-js/README.md [hunk 4]
```diff
@@ -1,3 +1,46 @@
...
+| Registering a user with the application.                                                | [Sign Up](https://aws-amplify.github.io/amplify-js/api/functions/aws_amplify.auth.signUp.html)                                        |
+| Confirming a registered, unauthenticated user using a confirmation code.                | [Confirm Sign Up](https://aws-amplify.github.io/amplify-js/api/functions/aws_amplify.auth.confirmSignUp.html)                         |
+| Resending a confirmation code form registration of an unauthenticated user.             | [Resend Sign Up Code](https://aws-amplify.github.io/amplify-js/api/functions/aws_amplify.auth.resendSignUpCode.html)                  |
+| Authenticating a user and establishing a user session.                                  | [Sign In](https://aws-amplify.github.io/amplify-js/api/functions/aws_amplify.auth.signIn.html)                                        |
+| Configuring an authentication flow type.                                                | [Auth Flows](https://docs.amplify.aws/react/build-a-backend/auth/switch-auth/)                                                        |
```
[the remark this answers is not in the document]
Comment (haverchuck) [reply packages/amazon-cognito-identity-js/README.md:16]: Done, where possible

File: packages/amazon-cognito-identity-js/README.md [hunk 5]
```diff
@@ -219,17 +262,20 @@ var attributePhoneNumber = new AmazonCognitoIdentity.CognitoUserAttribute(
...
 attributeList.push(attributePhoneNumber);
 
-userPool.signUp('username', 'password', attributeList, null, function(
-	err,
-	result
-) {
-	if (err) {
-		alert(err.message || JSON.stringify(err));
-		return;
+userPool.signUp(
```
[the remark this answers is not in the document]
Comment (haverchuck) [reply packages/amazon-cognito-identity-js/README.md:258]: Yes

File: packages/amazon-cognito-identity-js/README.md (hunk 1 above)
Comment (haverchuck) [reply packages/amazon-cognito-identity-js/README.md]: Copy/paste error: fixed

**Review by renebrandel — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Review by ashika112 — APPROVED** [review]

**Comment by kvramyasri7** [comment]
wow dustin!!!!!!!!!!

**Pull request merged**